### DISCUSSION TITLE
3D Touch for KP5L 1m cable is sufficient

### DISCUSSION THREAD

**Discussion by mechano** [mechano/Kingroon_KP5L_Marlin_Firmware#3] [Announcements]
1 meter of the standard cable that ships with 3D Touch or BL Touch is sufficient. 
The KP5L has a plug expressly for 3D Touch on an IC board into the frame where the X axis stepper is located.

Another IC board is under the bottom cover and you need to build 2 ribbons wuth plugs from the IC board to the mainboard.

![image](https://github.com/mechano/Kingroon_KP5L_Marlin_Firmware/assets/2874499/766f9c1a-5ab3-4ef9-aa2c-8992b2cb4ed6)

![image](https://github.com/mechano/Kingroon_KP5L_Marlin_Firmware/assets/2874499/07c88f14-ac0a-4952-8a75-0191f7d1ef54)

![IMG20231104161542](https://github.com/user-attachments/assets/7e91a1aa-9526-4f28-aa03-823a8604d79b)

**Comment by MrMores** [comment]
Hey, I just installed your firmware and am trying to wire up the 3D touch. I have 5 wires coming from the 3d touch; 3 of which I have plugged into the 3D touch pins on the main board and the remaining 2 on the Z- pins. The wires are in the right orientation and seem correct. After doing this the Z axis no longer wants to go down. And instead the 3D touch prong just pops in and out for a bit before just deciding that it has reached home even though it is 240mm up. Any idea what the issue might be?

**Reply by MrMores** [reply]
Just wanna add that when I had the remaining 2 wires on the Z+ pins then it was able to go down but it would not stop at the bottom and just force itself until I had to manually turn the whole printer off.

**Reply by mechano** [reply]
Sorry but me and thousands people around the world have a 3D Touch working with this firmware. 

All the problems come from bad wiring.
